### PR TITLE
Todo list export

### PR DESCRIPTION
**Pull request by FriendlyPopo**
User Adaram requested this in the developers guild. I decided it was an easy project to start looking at habit's code.

User asked for an export of tasks.

Not being familiar with most technologies involved, I'm not really expecting this code to be adopted. Instead, I'm really looking for feedback, tips and comments on how to do this better. 

Thanks in advance!

### CODE REVIEW

File: common/locales/en/settings.json [hunk 1]
```diff
@@ -104,5 +104,7 @@
     "benefits": "Benefits",
     "coupon": "Coupon",
     "couponPlaceholder": "Enter Coupon Code",
-    "couponText": "We sometimes have events and give out coupon codes for special gear. (eg, those who stop by our Wondercon booth)"
+    "couponText": "We sometimes have events and give out coupon codes for special gear. (eg, those who stop by our Wondercon booth)",
+    "currentTasks": "To-do list",
+    "exportAllActiveTodos": "All my to-dos:"
```
Comment (Alys) [inline common/locales/en/settings.json]: "To-Do" is the correct capitalisation (just for the text portion, not for the key).

**Comment by Alys** [comment]
Looks interesting at first glance! I'll try to make time in the next few days to go through the code in detail, if no one else does first. :)  

It would be very nice if the field headers and other text in the dataexport.todolist function could be translatable strings.

The date format should follow international standard ISO 8601 (e.g., 2015-03-09T13:07:04Z). Any decent spreadsheet application should then be able to convert that to the user's preferred format. It's also MUCH better for sorting chronologically than any format that doesn't have the year first. :)

Could you please rename this PR so that the title is more descriptive? That will help us when we're looking through the PR index screen, and will add useful keywords for title searches.

**Comment by Alys** [comment]
I can't test this because when I download your `develop` branch, I get other code changes, including the `User.findById(.leader, function(err, aborter){` line which is refusing to compile.

Can you resubmit this with just the changes for this PR?

One thing you might find helpful is always creating a new feature branch for every change you make. That way you can be working on several features at the same time without the code from each feature being inadvertently included in pull requests for the others. It's a lesson I learnt the hard way. :)

**Comment by Alys** [comment]
@Horacious I'm very interested in this change and your code looks good at a glance. Would you like advice about setting up branches to get this and other changes working together without conflicting with each other?

**Comment by FriendlyPopo** [comment]
Thanks for the offer, @Alys. I've been meaning to get a bit of git hands-on practice and tuts going before trying to contribute some more. However, we just had a new baby last week and I've been totally offline and braindead for the past days. Will come back in a week or two!

**Comment by lemoness** [comment]
Congratulations!!

**Comment by Adaram** [comment]
Congrats @Horacious that is awesome news!!+

**Comment by Adaram** [comment]
Hey all. Just wanted to follow up on this. Is there any way we can move the priority up? I'd really like to be able to get the To Do data, and copying/pasting off the in Habit Data Display Tool doesn't work very well - the information just doesn't play nice with excel  :(

Thanks in advance!

**Comment by Alys** [comment]
@Adaram As @Horacious says, he has a new baby to take care of. Important personal commitments should always take precedence over open-source volunteer work for HabitRPG.

**Comment by Adaram** [comment]
Yeah @Alys - totally understood and agreed.

**Comment by crookedneighbor** [comment]
@Horacious Are you still interested in working on this? If so, re-open this pull request.

**Comment by Alys** [comment]
Alternatively, @Horacious, let us know whether or not you'd mind if someone else worked with your code to implement this feature.

**Comment by FriendlyPopo** [comment]
No!  Of course I wouldn't mind at all!  By all means, reuse whatever you
want.

As you might have guessed, my new life has been keeping me outrageously
busy.  I can't even think of learning a new language.

Take care!

**Comment by Adaram** [comment]
@Horacious new babies tend to take an inordinate amount of time! It can be tiring, but don't blink! Before you know it they will be 21 and graduating from College (my son just did this!).

**Comment by Alys** [comment]
I've added this message to the [Aspiring Coders guild](https://habitrpg.com/#/options/groups/guilds/68d4a01e-db97-4786-8ee3-05d612c5af6f):

There's a pull request for a desirable feature that needs (I think) just a little more work before it can be released, but the original author isn't available to work on it any more: [Todo list export by Horacious - Pull Request #4862](https://github.com/HabitRPG/habitrpg/pull/4862). Would anyone like to finish it? Both you and Horacious would gain credit towards a contributor tier. Feel free to ask questions in the issue if you wish to.

**Pull request closed without merging**